install.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+# Copyright 2018 the Deno authors. All rights reserved. MIT license.
+# TODO(everyone): Keep this script simple and easily auditable.
+
+set -e
+
+deno_dir="$HOME/.deno/bin"
+deno_zip="${deno_dir}/deno.gz"
+deno_bin="${deno_dir}/deno"
+
+red="\033[31m"
+green="\033[32m"
+yellow="\033[33m"
+reset="\033[0m"
+
+if [ $# -eq 0 ]; then
+  # TODO: We could check the GitHub API for the latest release here.
+  # However, we need to parse the JSON response body. Perhaps
+  # this could be done with an existing installation of deno, jq,
+  # or node. And otherwise we ned to fall back to asking the user:
+  read -p "Which version do you want to install? " version
+else
+  version=$1
+fi
+
```

Comment:
@MarkTiedemann check how I determine the release url in [function release_url](https://github.com/chiefbiiko/deno-bash-installer/blob/63aada0db1792b20417bd4a4dcb5865a90cf8929/install.sh#L31) and [function main (l57-l59)](https://github.com/chiefbiiko/deno-bash-installer/blob/63aada0db1792b20417bd4a4dcb5865a90cf8929/install.sh#L57)
if u pass a tag that version is installed else the latest
this line `match=$(curl -sL $url | grep href | grep $filename | grep -o '/deno[^"]*')` has been quite reliable so far
the `grep`s work on mac and linux
just use a remix of `release_url` and that regex in `main`

Reply:
@chiefbiiko You're grepping the HTML of the release page? This could break anytime GitHub is updating their markup, so maybe it "has been quite reliable so far", but I wouldn't want to use that. Instead, I'd rather grep or json-parse the GitHub API response because, at the very least, the API has some stability guarantee.

For example: `curl -s https://api.github.com/repos/denoland/deno/releases/latest | grep tag_name | grep -o "v[0-9]\{1,\}\.[0-9]\{1,\}\.[0-9]\{1,\}"`

Then again, I'm not doing that on purpose to avoid complexity (see: [https://github.com/denoland/deno_std/issues/94#issuecomment-451981613](https://github.com/denoland/deno_std/issues/94#issuecomment-451981613)).

I'd be interested to know what @ry thinks about this. In general, we should decide whether we want to hard-code the version number or parse it from the GitHub API, and then implement that in all the different scripts.

Reply:
I'm okay with hardcoding it if we move this to deno_std. 

We are being very careful in our release process to make it as automated as possible. Currently the only manual steps for release are:
1. make a commit in deno bumping the version
2. push a tag
3. [update the CI scripts in deno_std](https://github.com/denoland/deno_std/commit/72b2e20f9da1d431ba9cac3247ee38ee82d5d8f4) 

Step 3 is annoying, but I want deno_std to be explicitly testing against a specific version. I don't want to also add a step 4 "Bump deno_install"

Another reason in favor of hard coding is that sometimes the Windows binaries are cut faster than the Linux or Mac binaries (because appveyor is faster) ... so for a short time there are no Linux and Mac releases of the "latest version". If we manually bumped it, it wouldn't be a problem.

Ideally I'd be able to script the action of bumping the deno_std version, so that the release process is minimal.

Reply:
> I don't want to also add a step 4 "Bump deno_install"

Agreed.

> Another reason in favor of hard coding is that sometimes the Windows binaries are cut faster than the Linux or Mac binaries (because appveyor is faster) ... so for a short time there are no Linux and Mac releases of the "latest version".

That's interesting! I didn't know that.

Looking at the `created_at` fields of the assets in the API response, "a short time" can actually mean up to 20 minutes in this case.

Hard-coding is fine with me. Alternatively, we could also get the last 2 releases from the API and use the first available `browser_download_url` for the OS the script is being run on.

```
$ curl -s https://api.github.com/repos/denoland/deno/releases?per_page=2 | grep browser_download_url
"browser_download_url": "https://github.com/denoland/deno/releases/download/v0.2.6/deno_linux_x64.gz"
"browser_download_url": "https://github.com/denoland/deno/releases/download/v0.2.6/deno_osx_x64.gz"
"browser_download_url": "https://github.com/denoland/deno/releases/download/v0.2.6/deno_win_x64.zip"
"browser_download_url": "https://github.com/denoland/deno/releases/download/v0.2.5/deno_linux_x64.gz"
"browser_download_url": "https://github.com/denoland/deno/releases/download/v0.2.5/deno_osx_x64.gz"
"browser_download_url": "https://github.com/denoland/deno/releases/download/v0.2.5/deno_win_x64.zip"
```

I guess it should be possible to implement this for the deno_std CI, too, right? If so, step 3 would not be needed anymore. :)

Reply:
Hard-coded the $version for now. I guess if we want to do it dynamically or move this code to deno_std, we can do so later.